### ISSUE TITLE
Broken user permissions, a user can view/change/delete other user's work and no privacy.

### ISSUE THREAD

**Issue by geeky-neo** [kiwitcms/Kiwi#2499]
HINT: try to reproduce bugs and errors at https://public.tenant.kiwitcms.org!
This is always running the latest version and will automatically submit
traceback and debugging information to us!

### Description of problem
Broken user permissions, a user can view/change/delete other user's work and no privacy.

### Steps to Reproduce

1. Create non-admin user Bob and Alice.
2. Create test runs Test_Run_1 and Test_Run_2.
3. Test_Run_2 is assigned to Bob and Test_Run_3 is assigned to Alice.

### Actual results
Bob is able to change the Test status of Test Case inside a Test Run assigned to Alice.

### Expected results
Bob should not be able to change/delete/view Alice's work/test plan/test run/test case.
Built-in privacy and access control of users/profiles by default.

**Comment by atodorov** [comment]
This is working as designed. The permissions system in general doesn't have the notion of object owners so permissions are granted on a object type/record type basis. In addition there's a second level of permissions called object-level permissions which allow you to limit scope per individual object -> https://kiwitcms.readthedocs.io/en/latest/admin.html#object-level-permissions

If you want true separation between various groups then you have multi-tenancy architecture on top of that -> https://github.com/kiwitcms/tenants/. 

>  Built-in privacy and access control of users/profiles by default.

In other words track user ownership of every possible object in the database, aka treat each user as a stand-alone tenant. This is not only impossible in the current architecture but also doesn't make a lot of sense for a group of people working together within a team.
